test/cache_invalidation_test.rb

```diff
@@ -46,14 +46,14 @@ def test_reload_invalidate_cached_objects
   def test_after_a_reload_the_cache_perform_as_expected
     Item.cache_has_many :associated_records, :embed => :ids
 
-    assert_equal [@baz, @bar], @record.associated_records
     assert_equal [@baz, @bar], @record.fetch_associated_records
+    assert_equal [@baz, @bar], @record.associated_records
```

Comment:
The key is that this test should catch regressions in [lib/identity_cache/cache_invalidation.rb](https://github.com/Shopify/identity_cache/blob/v0.3.1/lib/identity_cache/cache_invalidation.rb), but calling `@record.associated_records` before `@record.fetch_associated_records` would miss regressions there because then `@record.fetch_associated_records` would just delegate to `@record.associated_records`

Right, I just don't want to have `@record.fetch_associated_records` delegate to